R/vsc.R

```diff
@@ -7,6 +7,91 @@ homedir <- Sys.getenv(
 dir_watcher <- Sys.getenv("VSCODE_WATCHER_DIR", file.path(homedir, ".vscode-R"))
 request_file <- file.path(dir_watcher, "request.log")
 request_lock_file <- file.path(dir_watcher, "request.lock")
+settings_file <- file.path(dir_watcher, "settings.json")
+user_options <- names(options())
+
+load_settings <- function() {
+  if (!file.exists(settings_file)) {
+    return(FALSE)
+  }
+
+  rSettings <- c(
+    "plot.useHttpgd",
+    "workspaceViewer.showObjectSize",
+    "session.emulateRStudioAPI",
+    "session.levelOfObjectDetail",
+    "session.watchGlobalEnvironment",
+    "session.viewers.viewColumn.plot",
+    "session.viewers.viewColumn.browser",
+    "session.viewers.viewColumn.viewer",
+    "session.viewers.viewColumn.pageViewer",
+    "session.viewers.viewColumn.view",
+    "session.viewers.viewColumn.helpPanel "
+  )
+
+  # settings.json can result in errors if read via fromJSON
+  vsc_settings <- suppressWarnings(
+    unlist(
+      jsonlite::fromJSON(paste(readLines(settings_file), collapse = ""))
```

Comment:
Why not use `jsonlite::read_json()`? Could you provide a case where "settings.json can result in errors if read via fromJSON"?

Reply:
OK. Let me refine the code a bit so that naming conventions are more consistent.